analysis/ic50_analysis.py

```diff
@@ -0,0 +1,62 @@
+import json
+import sys
+from typing import Any
+
+from ic50 import *
+
+
+def main(argv: Any) -> None:
+    data = json.loads(argv[0])
+
+    response = []
+
+    for experiment in data:
+        response_value = {"experiment": experiment["experiment"], "status": "DONE"}
+        measurements = np.array(experiment['data'])
+
+        try:
+            find_outliers = experiment["find_outliers"]
+            if type(find_outliers) == str:
+                find_outliers = find_outliers.lower()
+                find_outliers = int(find_outliers == 'true')
+        except KeyError:
+            find_outliers = True
```

Comment:
Can you please explain it. Why `find_outliers` can be `str`? Why we need this `try/except` here?

Reply:
AFAIU the logic here is to support 3 cases: `find_outliers` is passed as normal `bool`, as a string with `"true` or `"false"` or not passed at all (defaults to `true`). My code should always pass normal bool (`true` or `false`), so `find_outliers = experiment["find_outliers"]` should be enough, but I decided that such generalization is harmless and decided not to comment these lines of code.

> Why we need this `try/except` here?

Actually I have a question regarding `try/except` here, too. Isn't it neater to write `experiment.get("find_outliers", true)`?

Reply:
Right, I didn't know which format will be passed, so included str just for safety. Regarding `get()` method, it return None if no such key existent. I chose `try/except` because thought, in the future we might want to return an exception if the format of json is incorrect. I don't think there's big of a difference.

Reply:
> Regarding `get()` method, it returns `None` if no such key existent.

[`get()`](https://docs.python.org/3/library/stdtypes.html#dict.get) has another argument that specifies the default value to be returned if no value is present. It defaults to `None`, but I suggest passing `True` as the default value.

Reply:
So we wouldn't need such exception? Sure, will do it this way